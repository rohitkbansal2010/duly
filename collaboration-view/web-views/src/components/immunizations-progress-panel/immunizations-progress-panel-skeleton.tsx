import { DulyLoader } from '@components/duly-loader';
import { IMMUNIZATIONS_LOADER_WIDTH } from '@constants';

import styles from './immunizations-progress-panel.scss';

export const ImmunizationsProgressPanelSkeleton = () =>
  (
    <div className={styles.progressPanelSkeleton}>
      <DulyLoader width={IMMUNIZATIONS_LOADER_WIDTH} />      
    </div>
  );
