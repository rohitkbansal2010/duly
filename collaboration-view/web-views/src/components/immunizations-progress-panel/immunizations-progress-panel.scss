@import "~@styles/typography.scss";
@import "~@styles/themes.scss";

.progressPanel {
  &Container {
    display: flex;
    margin-bottom: 1.5rem;
    padding: 1.5rem 1.25rem;
    background: $white;
    border-radius: 0.5rem;
  }

  &Status {
    display: flex;
    align-items: center;
    width: 54%;

    &Icon {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 4.8rem;
      height: 4.8rem;
      margin-right: 1.25rem;
      border-radius: 50%;

      img {
        height: 2.75rem;
      }

      &Completed {
        background: rgba(120, 190, 33, 0.3);
      }

      &Almost {
        background: rgba(0, 165, 223, 0.3);
      }

      &LaggingBehind {
        background: rgba(255, 106, 20, 0.3);
      }
    }

    &Titles {
      @include nexa-regular-highlight;

      display: flex;
      flex-direction: column;
      font-size: var(--font-size-20px-4px);
      line-height: var(--line-height-22px-4px);
    }

    &Title {
      @include nexa-heavy-highlight;
      font-size: var(--font-size-32px-4px);
      line-height: var(--line-height-34px-4px);
    }
  }

  &Progress {
    display: flex;
    align-items: center;
    width: 23%;

    &Doughnut {
      width: 5.3rem;
      height: 5.3rem;
      margin-right: 1.25rem;
      border-radius: 50%;
    }

    &Titles {
      @include inter-text-default-bold;
      display: flex;
      flex-direction: column;
      font-size: var(--font-size-16px-4px);
      line-height: var(--line-height-28px-4px);
      text-transform: capitalize;
    }

    &Title {
      @include inter-text-xs-bold;
      color: $dark-gray;
      font-size: var(--font-size-14px-4px);
      line-height: 1.5rem;
      text-transform: uppercase;
    }
  }

  &Recommended {
    display: flex;
    align-items: center;
    width: 23%;

    &Icon {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 4.8rem;
      height: 4.8rem;
      margin-right: 0.875rem;
      background: $light-blue;
      border-radius: 50%;

      img {
        height: 3.6rem;
      }
    }

    &Titles {
      @include inter-text-default-bold;
      display: flex;
      flex-direction: column;
      font-size: var(--font-size-16px-4px);
      line-height: var(--line-height-28px-4px);
      text-transform: uppercase;
    }

    &Title {
      @include inter-text-xs-bold;
      color: $dark-gray;
      font-size: var(--font-size-14px-4px);
      line-height: 1.5rem;
    }
  }

  &Skeleton {
    position: relative;
    height: 8.3rem;
    margin-bottom: 1.5rem;
    background: $white;
    border-radius: 0.5rem;
  }
}
