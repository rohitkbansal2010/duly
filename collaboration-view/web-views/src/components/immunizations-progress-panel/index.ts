export { ImmunizationsProgressPanel } from './immunizations-progress-panel';
export { ImmunizationsProgressPanelSkeleton } from './immunizations-progress-panel-skeleton';
