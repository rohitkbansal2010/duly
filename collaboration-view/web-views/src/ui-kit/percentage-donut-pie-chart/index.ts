export * from './percentage-donut-pie-chart';
