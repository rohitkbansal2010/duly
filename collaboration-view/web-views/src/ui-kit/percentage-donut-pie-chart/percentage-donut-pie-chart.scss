@import "~@styles/colors.scss";
@import "~@styles/typography.scss";

.pieChart {
  position: relative;

  &Content {
    @include inter-h3-bold;

    &Percent {
      color: $light-violet;
      vertical-align: middle;
    }

    &Container {
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }
}
