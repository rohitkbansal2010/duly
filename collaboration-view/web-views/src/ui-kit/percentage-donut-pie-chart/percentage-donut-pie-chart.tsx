import styles from './percentage-donut-pie-chart.scss';

type PieChartType = {
  percent: number;
};

const RADIUS_INNER_CIRCLE = 23;
const RADIUS_OUTER_CIRCLE = 30;
const RADIUS_VIEWBOX = 36;
const INNER_STROKE_WIDTH = 5;
const OUTER_STROKE_WIDTH = INNER_STROKE_WIDTH * 2;
const DIAMETER_OUTER_CIRCLE = RADIUS_OUTER_CIRCLE * 2;
const DIAMETER_VIEWBOX = RADIUS_VIEWBOX * 2;
const MAX_PERCENT = 100;

export const PercentageDonutPieChart = ({ percent }: PieChartType) => {
  const dashCount = ((percent * Math.PI * DIAMETER_OUTER_CIRCLE) / MAX_PERCENT);
  const transform = `rotate(-90) translate(-${DIAMETER_VIEWBOX})`;
  
  return (
    <div className={styles.pieChart}>
      <svg width="100%" height="100%" viewBox={`0 0 ${DIAMETER_VIEWBOX} ${DIAMETER_VIEWBOX}`}>
        <circle
          r={RADIUS_OUTER_CIRCLE}
          cx={RADIUS_VIEWBOX}
          cy={RADIUS_VIEWBOX}
          fill="transparent"
          stroke="#79065a"
          strokeWidth={INNER_STROKE_WIDTH}
        />
        <circle
          r={RADIUS_OUTER_CIRCLE}
          cx={RADIUS_VIEWBOX}
          cy={RADIUS_VIEWBOX}
          fill="transparent"
          stroke="#c5299b"
          strokeWidth={OUTER_STROKE_WIDTH}
          strokeDasharray={`${dashCount} ${Math.PI * DIAMETER_OUTER_CIRCLE}`}
          transform={transform}
        />
        <circle
          r={RADIUS_INNER_CIRCLE}
          cx={RADIUS_VIEWBOX}
          cy={RADIUS_VIEWBOX}
          fill="#79065a"
        />
      </svg>
      <div className={styles.pieChartContentContainer}>
        <span className={styles.pieChartContent}>
          <span className={styles.pieChartContentPercent}>{percent}</span>
        </span>
      </div>
    </div>
  );
};
