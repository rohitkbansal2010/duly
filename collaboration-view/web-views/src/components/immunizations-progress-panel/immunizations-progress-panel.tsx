import { PROGRESS_PANEL } from '@constants';
import { immunizationLightBlueIcon } from '@icons';
import { ImmunizationsProgress } from '@types';
import { PercentageDonutPieChart } from '@ui-kit/percentage-donut-pie-chart';

import styles from './immunizations-progress-panel.scss';

export const ImmunizationsProgressPanel =
  ({ percentageCompletion, recommendedGroupNumber, completionStatus }: ImmunizationsProgress) => {
    const iconWrapperClasses = []
      .concat(styles.progressPanelStatusIcon)
      .concat(styles[`progressPanelStatusIcon${completionStatus}`])
      .join(' ');

    return (
      <div className={styles.progressPanelContainer}>
        <div className={styles.progressPanelStatus}>
          <div className={iconWrapperClasses}>
            <img src={PROGRESS_PANEL[completionStatus].icon} alt={`${completionStatus} status`}/>
          </div>
          <div className={styles.progressPanelStatusTitles}>
            <span className={styles.progressPanelStatusTitle}>
              {PROGRESS_PANEL[completionStatus].title}
            </span>
            <span>
              {PROGRESS_PANEL[completionStatus].additional}
            </span>
          </div>
        </div>
        <div className={styles.progressPanelProgress}>
          <div className={styles.progressPanelProgressDoughnut}>
            <PercentageDonutPieChart percent={percentageCompletion}/>
          </div>
          <div className={styles.progressPanelProgressTitles}>
            <span className={styles.progressPanelProgressTitle}>progress</span>
            <span>
              {`${percentageCompletion}% completed`}
            </span>
          </div>
        </div>
        <div className={styles.progressPanelRecommended}>
          <div className={styles.progressPanelRecommendedIcon}>
            <img src={immunizationLightBlueIcon} alt="recommended immunizations"/>
          </div>
          <div className={styles.progressPanelRecommendedTitles}>
            <span className={styles.progressPanelRecommendedTitle}>recommended</span>
            <span>
              {recommendedGroupNumber} immunization{recommendedGroupNumber === 1 ? '' : 's'}
            </span>
          </div>
        </div>
      </div>
    );
  };
