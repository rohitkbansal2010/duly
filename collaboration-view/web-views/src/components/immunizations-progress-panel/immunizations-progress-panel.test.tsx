import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PROGRESS_PANEL } from '@constants';
import { immunizationsWidgetMockData } from '@mock-data';
import { ImmunizationsProgress } from '@types';
import { PercentageDonutPieChart } from '@ui-kit/percentage-donut-pie-chart';

import { ImmunizationsProgressPanel } from './immunizations-progress-panel';

const mockProgress: ImmunizationsProgress =
  immunizationsWidgetMockData.progress;

describe('ImmunizationsProgressPanel', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = shallow(
      <ImmunizationsProgressPanel {...mockProgress} />
    );
  });

  it('should render component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should contains right titles and icon', () => {
    const { title, additional, icon } = PROGRESS_PANEL[mockProgress.completionStatus];

    expect(component.text().includes(title)).toBeTruthy();
    expect(component.text().includes(additional)).toBeTruthy();
    expect(component.findWhere(n => 
      n.type() === 'img' && n.props().src === icon)).toHaveLength(1);
  });

  it('should contains right ending for recommended immunizations', () => {
    expect(component.text().includes('immunizations')).toBeTruthy();

    component = shallow(
      <ImmunizationsProgressPanel
        {...mockProgress}
        recommendedGroupNumber={1}
      />
    );

    expect(component.text().includes('immunizations')).toBeFalsy();
    expect(component.text().includes('immunization')).toBeTruthy();
  });

  it('should contains right props for donut chart', () => {
    expect(component.find(PercentageDonutPieChart)
      .props().percent === mockProgress.percentageCompletion).toBeTruthy();
  });
});
