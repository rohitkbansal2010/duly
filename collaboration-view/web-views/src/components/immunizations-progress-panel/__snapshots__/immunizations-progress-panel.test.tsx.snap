// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImmunizationsProgressPanel should render component 1`] = `
<div
  className="progressPanelContainer"
>
  <div
    className="progressPanelStatus"
  >
    <div
      className="progressPanelStatusIcon progressPanelStatusIconAlmost"
    >
      <img
        alt="Almost status"
        src="status-almost-light-blue.svg"
      />
    </div>
    <div
      className="progressPanelStatusTitles"
    >
      <span
        className="progressPanelStatusTitle"
      >
        You’re almost there!
      </span>
      <span>
        Few more and you’re all set!
      </span>
    </div>
  </div>
  <div
    className="progressPanelProgress"
  >
    <div
      className="progressPanelProgressDoughnut"
    >
      <PercentageDonutPieChart
        percent={75}
      />
    </div>
    <div
      className="progressPanelProgressTitles"
    >
      <span
        className="progressPanelProgressTitle"
      >
        progress
      </span>
      <span>
        75% completed
      </span>
    </div>
  </div>
  <div
    className="progressPanelRecommended"
  >
    <div
      className="progressPanelRecommendedIcon"
    >
      <img
        alt="recommended immunizations"
        src="immunization-light-blue.svg"
      />
    </div>
    <div
      className="progressPanelRecommendedTitles"
    >
      <span
        className="progressPanelRecommendedTitle"
      >
        recommended
      </span>
      <span>
        4
         immunization
        s
      </span>
    </div>
  </div>
</div>
`;
